### PR TITLE
Update Stack to have an `as` prop to specify the HTML element to rend…

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

All for the `Stack` component to be rendered as any HTML element

## Test Steps

Updated tests to cover usage

1. Checkout branch - `git checkout stack-as-any-element`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the stack storybook documentation

## Author Checklist 

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed relevant documentation has been updated/added.
- [x] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

### CODE REVIEW

**Comment by nschubach** [comment]
I only worry that this will cause anything that has `as` to inflate to support features on a tag by tag basis.

**Review by nschubach — APPROVED** [review]

**Comment by matthewroach** [comment]
@nschubach - I hear your concern, and for that I have updated the documentation to give more explanation on what the `as` props intended use case is

File: src/components/stack/index.stories.mdx [hunk 1]
```diff
@@ -12,6 +12,17 @@ It allows for a both a column or row layout. In addition,
...
+## as
+
+The `Stack` will default to use the HTML `div` element for rendering. There will be times
+where the `Stack` will be used for sectioning and the `div` element does not provide semantic
+meaning, to avoid having to use additional markup to bring back semantics use the `as` prop to
+render the `Stack` as a different HTML element. By using the `as` property you are then required
+to ensure all the correct attributes for a given tag are still passed through.
+
+**Note:** The `as` prop's primary use case if for sectioning. There is likely a component
+better suited for your needs. For example: `Link` component should be used for anchor's.
```
Comment (nschubach) [inline src/components/stack/index.stories.mdx:24]:
```suggestion
better suited for your needs. For example, the `Link` component should be used for anchors.
```

**Review by nschubach — APPROVED** [review]
Approving, minor grammatical concern.

**Pull request merged**